dht/range_streamer.cc

```diff
@@ -269,7 +269,8 @@ future<> range_streamer::stream_async() {
                 break;
             } catch (...) {
                 logger.warn("{} failed to stream. Will retry in {} seconds ...", _description, sleep_time);
-                sleep_abortable(std::chrono::seconds(sleep_time)).get();
+                _abort_source.check();
```

Comment:
doesn't sleep_abortable() already check this?

Reply:
It does not, AFAICT.

Reply:
`sleep_abortable()` constructs a `sleeper` object which calls `as.subscribe()` and if that returns a disengaged optional then it won't arm the timer but resolve with an exception instead. `abort_source::subscribe()` checks for `abort_requested()` and if so returns a disengaged optional.

So it seems that `sleep_abortable()` already aborts immediately when the abort_source is aborted, so calling `check()` before it seems not necessary.

Reply:
Yes. I looked at it but misread it (and the behavior is expected). I'll remove the checks.